code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/legionnaire.dm

```diff
@@ -0,0 +1,296 @@
+	var/turf/T = get_step(get_turf(src), dir_to_target)
+	for(var/i in 1 to 4)
+		new /obj/effect/temp_visual/dragon_swoop/legionnaire(T)
+		T = get_step(T, dir_to_target)
+	playsound(src,'sound/magic/demon_attack1.ogg', 200, 1)
+	visible_message("<span class='boldwarning'>[src] prepares to charge!</span>")
+	addtimer(CALLBACK(src, .proc/legionnaire_charge_2, dir_to_target, 0), 5)
+	return
+	
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/legionnaire_charge_2(var/move_dir, var/times_ran)
+	if(times_ran < 4)
+		var/turf/T = get_step(get_turf(src), move_dir)
+		if(ismineralturf(T))
+			var/turf/closed/mineral/M = T
+			M.gets_drilled()
+		if(!T.density)
+			for(var/obj/structure/window/W in T.contents)
+				return
+			for(var/obj/machinery/door/D in T.contents)
+				return
+			forceMove(T)
+			playsound(src,'sound/effects/bang.ogg', 200, 1)
+			var/list/hit_things
+			var/throwtarget = get_edge_target_turf(src, move_dir)
+			for(var/mob/living/L in T.contents - hit_things - src)
+				if(!faction_check_mob(L))
+					hit_things += L
+					visible_message("<span class='boldwarning'>[src] grasps [L] and throws them with much force!</span>")
+					to_chat(L, "<span class='userdanger'>[src] grabs you and throws you with much force!</span>")
+					L.safe_throw_at(throwtarget, 10, 1, src)
+					L.Paralyze(20)
+					L.adjustBruteLoss(50)
+			addtimer(CALLBACK(src, .proc/legionnaire_charge_2, move_dir, (times_ran + 1)), 2)
+	else
+		return
+		
+/mob/living/simple_animal/hostile/asteroid/elite/legionnairehead
+	name = "legionnaire head"
+	desc = "The legionnaire's head floating by itself.  One shouldn't get too close, though once it sees you, you really don't have a choice."
+	icon_state = "legionnaire_head"
+	icon_living = "legionnaire_head"
+	icon_aggro = "legionnaire_head"
+	icon_dead = "legionnaire_dead"
+	icon_gib = "syndicate_gib"
+	maxHealth = 80
+	health = 80
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	attack_verb_continuous = "bites at"
+	attack_verb_simple = "bite at"
+	attack_sound = 'sound/effects/curse1.ogg'
+	throw_message = "simply misses"
+	speed = 0
+	move_to_delay = 2
+	del_on_death = 1
+	deathmessage = "crumbles away!"
+	faction = list()
+	ranged = 0
+	var/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/body = null
+	
+/mob/living/simple_animal/hostile/asteroid/elite/legionnairehead/death()
+	. = ..()
+	if(body)
+		body.onHeadDeath()
+	return
+		
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/head_detach(target)
+	ranged_cooldown = world.time + 5
+	if(has_head)
+		has_head = FALSE
+		icon_state = "legionnaire_headless"
+		icon_living = "legionnaire_headless"
+		icon_aggro = "legionnaire_headless"
+		visible_message("<span class='boldwarning'>[src]'s head flies off!</span>")
+		var/mob/living/simple_animal/hostile/asteroid/elite/legionnairehead/newhead = new /mob/living/simple_animal/hostile/asteroid/elite/legionnairehead(loc)
```

Comment:
All these 'spawn child mob and give it our current target' stuff could be extracted into a helper on base type.

Reply:
The only thing about that is there's some case-by-case stuff happening depending on the elite.  For example, the broodmother adds all of her kids to a list, whereas the legionnaire has a reference variable to his head, and his summon mechanic also adjusts how much damage the head does, and herald's mirror doesn't do anything regarding targets.